### ISSUE TITLE
Unable to detect the web framework in use when using angular app within nx monorepo

### ISSUE THREAD

**Issue by kedevked** [firebase/firebase-tools#6049]
### [REQUIRED] Environment info

**firebase-tools: 12.4.1**

**Platform: Windows wsl**

### [REQUIRED] Steps to reproduce

- create an nx monorepo with an angular app: 

    npx create-nx-workspace@latest 

✔ Where would you like to create your workspace? · org
✔ Which stack do you want to use? · angular
✔ Standalone project or integrated monorepo? · integrated
✔ Application name · org
✔ Default stylesheet format · scss
✔ Would you like to use Standalone Components in your application? · Yes
✔ Would you like to add routing? · Yes
✔ Enable distributed caching to make your CI faster · No

    npm install @angular/fire

    npx nx g @angular/fire:ng-add

    firebase deploy

### [REQUIRED] Expected behavior

expect the app to be deployed to firebase

### [REQUIRED] Actual behavior

throws error

> Unable to detect the web framework in use

**Comment by robinstaudinger** [comment]
I'm facing the same issue with NextJS and NX

**Comment by borisyordanov** [comment]
Monorepos are not officially supported https://github.com/firebase/firebase-tools/issues/653
You might be able to create a hack to get it working, but from my experience you're better of looking elsewhere for your monorepo hosting needs

**Comment by kedevked** [comment]
Given the popularity of `nx` within the JavaScript ecosystem tooling, I would be surprised if `nx` is officially not supported

**Comment by ezzabuzaid** [comment]
This line - [frameworks/angular/index.ts](https://github.com/firebase/firebase-tools/blob/05ed2aa58ee35c2979e875db71a035b68250f1c5/src/frameworks/angular/index.ts#L39C6-L39C6)

It requires angular.json, if we can have that as parameter it might work!

**Comment by Niklas3011** [comment]
Is there an alternative to firebase which is supported for an Angular App within a NX Monorepo?

**Comment by SvenBudak** [comment]
Since Angular 17 and the SSR update, we have been using SSR and Firebase hosting. However, this is currently not possible as all our apps are in an NX monorepo. We are currently separating all apps into their own repositories, but this makes maintenance much more difficult due to common libraries. We would really appreciate a solution that is directly integrated into the Firebase CLI.

**Comment by Crocsx** [comment]
Yeah, same here, it took me to long to find this thread, If this is not planned to be supported, I would at least expect some clear info in the doc stating that mono repo are not supported.

**Comment by AlmaniaM** [comment]
We're using Angular 18 SSR with NX v19 and @alexdabast's solution is working but not sustainable. The project is recognized as a Vite project and it does not work. Are there any plans to support NX monorepos with Angular or any other project?

**Comment by alexdabast** [comment]
@AlmaniaM we are not using firebase "App Hosting" feature and it supports everything
We also move to pnpm who is also supported by app hosting

**Comment by AlmaniaM** [comment]
> @AlmaniaM we are now using firebase "App Hosting" feature and it supports everything
> We also move to pnpm who is also supported by app hosting

Interesting. I've been trying to get App Hosting to work but It keeps failing to deploy on the last step with no helpful logging either.

**Comment by worthy7** [comment]
> [@AlmaniaM](https://github.com/AlmaniaM) we are now using firebase "App Hosting" feature and it supports everything We also move to pnpm who is also supported by app hosting

So does app hosting somehow support having multiple angular projects in one workspace? Can you elaborate a bit?
